packages/analytics-browser/test/storage/local-storage.test.ts

```diff
@@ -12,4 +13,26 @@ describe('local-storage', () => {
     const localStorage = new LocalStorage();
     expect(await localStorage.isEnabled()).toBe(false);
   });
+
+  test('should drop events when set more than 1000 events without logging', async () => {
+    const localStorage = new LocalStorage<number[]>();
+
+    await localStorage.set('storage-key', new Array<number>(1001).fill(1));
+    const value = await localStorage.get('storage-key');
+
+    expect(value!.length).toBe(1000);
```

Comment:
would prefer to use optional chaining (`?`) vs non-null assertion (`!`) as it acknowledges that the object can be `undefined` as opposed to overruling type rules

```suggestion
    expect(value?.length).toBe(1000);
```

would probably yield to more meaning full error messages too I think